control-board/src/motion/robot_controller.rs

```diff
@@ -229,4 +239,320 @@ impl<'a> BodyVelocityController<'a> {
     pub fn get_control_debug_telem(&self) -> ControlDebugTelemetry {
         self.debug_telemetry
     }
+}
+
+impl<'a> ParameterInterface for BodyVelocityController<'a> {
+    fn processes_cmd(&self, param_cmd: &ParameterCommand) -> bool {
+        return self.has_name(param_cmd.parameter_name);
+    }
+
+    fn has_name(&self, param_name: ParameterName::Type) -> bool {
+        return match param_name {
+            RC_BODY_VEL_LIMIT | RC_BODY_ACC_LIMIT | RC_WHEEL_ACC_LIMIT => true,
+            VEL_PID_X | VEL_PID_Y | ANGULAR_VEL_PID_Z => true,
+            VEL_CGKF_ENCODER_NOISE | VEL_CGKF_PROCESS_NOISE | VEL_CGKF_GYRO_NOISE | VEL_CGFK_INITIAL_COVARIANCE | VEL_CGKF_K_MATRIX => true,
+            _ => false,
+        }
+    }
+
+
+    fn apply_command(&mut self, param_cmd: &ParameterCommand) -> Result<ParameterCommand, ParameterCommand> {
+        let mut reply_cmd = param_cmd.clone();
+
+        // if we haven't been given an actionable command code, ignore the call
+        if !(param_cmd.command_code == PCC_READ || param_cmd.command_code == PCC_WRITE) {
+            defmt::warn!("asked to apply a command with out and actional command code");
+            return Err(reply_cmd);
+        }
+
+        // if we've been asked to apply a command we don't have a key for it
+        // error out
+        if !self.has_name(param_cmd.parameter_name) {
+            defmt::warn!("asked to apply a command with a parameter name not managed by this module");
+            reply_cmd.command_code = PCC_NACK_INVALID_NAME;
+            return Err(*param_cmd);
+        }
+
+        if param_cmd.command_code == PCC_READ {
+            match param_cmd.parameter_name {
+                VEL_PID_X => {
+                    // set the type
+                    reply_cmd.data_format = PID_LIMITED_INTEGRAL_F32;
+
+                    // readback the data
+                    #[allow(non_snake_case)] // K is the mathematical symbol for this matrix
```

Comment:
I am cool with changing the name to like pid_gain to get rid of these and add clarity

Reply:
Kicking this one to #45 as it affects the CGKF as well and other bits of motion that need a refactor anyway